src/Core/Scheduler/Providers/TileProvider.js

```diff
@@ -32,6 +34,110 @@ TileProvider.prototype.preprocessDataLayer = function preprocessLayer(layer, vie
         throw new Error(`Cannot init tiled layer without schemeTile for layer ${layer.id}`);
     }
 
+    layer.addColorLayer = (colorLayer, addToView = true) => {
```

Comment:
I think this isn't the best place to overload the tileLayer (with `addColorLayer`....) the layer here because `TileProvider` provides tiles and pre-process tiles.
We could imagine a separate file that would be used in `createGlobe` and `createPlanar`

Reply:
That's what I thought initially, but actually adding `addColorLayer` and others functions in `TileProvider` makes sense: these features works when using TileMesh + LayeredMaterial which is exactly what `TileProvider` builds.

Reply:
`tileProvider` provides the data (image, elevation...) container and not the data.
A movie screen provider doesn't handle movies.
`tileProvider` should not handle anything other than tiles

Reply:
I think you got it backwards.
`TileProvider` doesn't provide the color layer support.
TileProvider simply knows that layers using the `tile` protocol support color/elevation/feature layer so it adds the `addXXXLayer` method, that's it.

Reply:
`TileProvider` should only provide tiles and provide the first tiles during the preprocess.
I don't see why a tile provider add functions to add another layer.

I find it too specific because why we could not add a feature / color layer to anything other than tiles

Reply:
Let's imagine we add a new feature to itowns: colorize a point cloud based on a WMS service.

To express this new feature, the PointCloudProvider would add a addColorLayer to a pointcloud layer.

This way, layers have specific methods for all addXXXLayer they support + specific documentation coming from jsdoc virtual comment.
And it seems clearer as a user, than a generic `addLayer` interface that will alwyas be available but with a chaning behavior depending on the underlying layer.

Reply:
The code considers that the tile has a `layeredMaterial`. The tile will not have a layered material all the time and the tile should not be constrained to a single material. This code is more specific to the material than to the tile. The only role of `tileProvider` is to provide mesh.


How do you add/remove/display the same layer wms on tile and point cloud?

Reply:
> The code considers that the tile has a layeredMaterial

No this code assumes that tiles produced by TileProvider supports color layers.

> The tile will not have a layered material all the time and the tile should not be constrained to a single material. 

This has been the case since the beginning of iTowns and I'm not aware of any plans to remove the mutual dependency.
So unless I've missed something I think we can reasonnably assume that TileProvider / TileMesh and LayeredMaterial are heavily linked.

> The only role of tileProvider is to provide mesh.

Did you read the code I added in TileProvider? This doesn't change TileProvider role, it's simply about extending layers using the `tile` protocol to inform the user they can add color (and other) layers to it.
Would it make you happy if I moved the `addXXXLayer`/`removeXXXLayer` code to `DefaultGeometryLayers.js`?

Reply:
it's better to place in `DefaultGeometryLayers.js`, and in addition to factoring common things in `createGlobe` and `createPlanar`